### PR TITLE
Fix for the SF2 with no "default" patch

### PR DESCRIPTION
**Pull request by kdeng00**
Using the instructions from #323 to test the fix for this PR, I was able to reproduce the problem and provide a fix. Let me know if there is anything else that is left from the issue.

### CODE REVIEW

File: plugins/sf2_player/sf2_player.cpp [hunk 1]
```diff
@@ -267,6 +268,46 @@ void sf2Instrument::loadFile( const QString & _file )
...
+			fluid_preset_t *pCurPreset = &preset;
+#else
+			fluid_preset_t *pCurPreset;
+#endif
+
+			while ( ( pCurPreset = fluid_sfont_iteration_next_wrapper( pSoundFont, pCurPreset ) ) ) {
+#ifdef CONFIG_FLUID_BANK_OFFSET
+				iBank += iBankOff;
+#endif
+				if ( pCurPreset && !foundFirstPreset ) {
```
Comment (PhysSong) [inline plugins/sf2_player/sf2_player.cpp]: * Checking for `pCurPreset` looks redundant because the `while` statement already checks for it.
* Can you use an `if` statement and a `break` statement instead of this double-loop? The current implementation looks very inefficient.

File: plugins/sf2_player/sf2_player.cpp [hunk 2]
```diff
@@ -267,6 +268,46 @@ void sf2Instrument::loadFile( const QString & _file )
...
+			fluid_preset_t preset;
+			fluid_preset_t *pCurPreset = &preset;
+#else
+			fluid_preset_t *pCurPreset;
+#endif
+
+			while ( ( pCurPreset = fluid_sfont_iteration_next_wrapper( pSoundFont, pCurPreset ) ) ) {
+#ifdef CONFIG_FLUID_BANK_OFFSET
+				iBank += iBankOff;
+#endif
```
Comment (PhysSong) [inline plugins/sf2_player/sf2_player.cpp]: I think `iBankOff` should be added to `iBank` after getting the value of `iBank`.

File: plugins/sf2_player/sf2_player.cpp [hunk 3]
```diff
@@ -184,6 +184,7 @@ sf2Instrument::sf2Instrument( InstrumentTrack * _instrument_track ) :
 
 	InstrumentPlayHandle * iph = new InstrumentPlayHandle( this, _instrument_track );
 	Engine::mixer()->addPlayHandle( iph );
+
```
Comment (PhysSong) [inline plugins/sf2_player/sf2_player.cpp]: Could you revert unnecessary whitespace changes(this includes new lines) if you can?

**Comment by kdeng00** [comment]
Hey @PhysSong, 
Thanks for the feedback and I made the changes. However, there is one test that is failing for Mac OS X and I can't seem to understand why it is failing.

File: plugins/sf2_player/sf2_player.cpp (hunk 1 above)
Comment (kdeng00) [reply plugins/sf2_player/sf2_player.cpp]: Made the fix https://github.com/LMMS/lmms/pull/5084/commits/37e151b5fc1e7661d153a782e449e6db91227257

File: plugins/sf2_player/sf2_player.cpp (hunk 2 above)
Comment (kdeng00) [reply plugins/sf2_player/sf2_player.cpp]: Made the fix https://github.com/LMMS/lmms/pull/5084/commits/37e151b5fc1e7661d153a782e449e6db91227257

File: plugins/sf2_player/sf2_player.cpp (hunk 3 above)
Comment (kdeng00) [reply plugins/sf2_player/sf2_player.cpp]: Made the fix https://github.com/LMMS/lmms/pull/5084/commits/37e151b5fc1e7661d153a782e449e6db91227257

**Comment by zonkmachine** [comment]
> However, there is one test that is failing for Mac OS X and I can't seem to understand why it is failing.

I think it's Homebrew acting up a bit. Restarting the Mac build a couple of times worked.

**Comment by kdeng00** [comment]
Sounds good, I would have thought that since Mac OS was POSIX compliant that it would have just been compatible but hombrew tool is something else.
